mypy/semanal.py

```diff
@@ -2366,11 +2374,19 @@ def fail_typeddict_arg(self, message: str,
 
     def build_typeddict_typeinfo(self, name: str, items: List[str],
                                  types: List[Type]) -> TypeInfo:
-        mapping_value_type = join.join_type_list(types)
         fallback = (self.named_type_or_none('typing.Mapping',
```

Comment:
Maybe I am missing something, but ``TypedDict`` is special-cased in ``cheker.check_indexed_assignment``, however ``del d['a']`` indeed goes through just checking for ``__delitem__`` and the latter is looked up on the fallback. Maybe we can just special-case this too? ``TypedDict``s at runtime are just ``dict``s and have methods like ``update`` that can't be accessed on ``Mapping``. Also ``TypedDict`` with ``total=False`` can support item deletion.

Anyway, it looks like both ``Mapping`` and ``Dict`` fallbacks require some special-casing, and it is not something where I have a strong opinion.